confdgnmi/src/confd_gnmi_api_adapter.py

```diff
@@ -0,0 +1,348 @@
+import xml.etree.ElementTree as ET
+from socket import socket
+
+import _confd
+import sys
+# TODO replace maapi_low with high level MAAPI
+from _confd import maapi as maapi_low
+from confd import maapi, maagic
+
+from confd_gnmi_adapter import GnmiServerAdapter
+from confd_gnmi_common import *
+
+log = logging.getLogger('confd_gnmi_api_adapter')
+
+
+class GnmiConfDApiServerAdapter(GnmiServerAdapter):
+    confd_debug_level = _confd.TRACE
+    confd_addr: str = '127.0.0.1'
+    confd_port: int = _confd.CONFD_PORT
+
+    def __init__(self):
+        self.addr: str = ""
+        self.port: int = 0
+        self.username: str = ""
+        self.password: str = ""
+        self.maapisock = None
+        self.mp_inst = None
+
+    @staticmethod
+    def set_confd_debug_level(level):
+        if level == "debug":
+            GnmiConfDApiServerAdapter.confd_debug_level = _confd.DEBUG
+        elif level == "trace":
+            GnmiConfDApiServerAdapter.confd_debug_level = _confd.TRACE
+        elif level == "proto":
+            GnmiConfDApiServerAdapter.confd_debug_level = _confd.PROTO_TRACE
+        elif level == "silent":
+            GnmiConfDApiServerAdapter.confd_debug_level = _confd.SILENT
+        else:
+            log.warning("Unknow confd debug level %s", level)
+
+    @staticmethod
```

Comment:
only one confd level is allowed for now (by command line options), but added to `connect` method so it can be changed for the instance, the class variables are mostly holders for default values.